File: src/main/java/org/cubewhy/celestial/utils/GitUtils.java
package org.cubewhy.celestial.utils;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public final class GitUtils {
    private GitUtils() {
    }

    public static final Properties info = new Properties();

    static {
        try {
            try (InputStream inputStream = GitUtils.class.getClassLoader().getResourceAsStream("git.properties")) {
                if (inputStream != null) {
                    info.load(inputStream);
                }
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public static String getBuildVersion() {
        return info.getProperty("git.build.version");
    }

    public static String getBranch() {
        return info.getProperty("git.branch");
    }

    public static String getRemote() {
        return info.getProperty("git.remote.origin.url");
    }

    public static String getCommitId(boolean shortID) {
        return shortID ? info.getProperty("git.commit.id.abbrev") : info.getProperty("git.commit.id");
    }

    public static String getBuildUser() {
        return info.getProperty("git.build.user.name");
    }
}

File: src/main/kotlin/org/cubewhy/celestial/utils/GitUtils.kt
package org.cubewhy.celestial.utils

import java.io.IOException
import java.util.*

object GitUtils {
    val info: Properties = Properties()

    init {
        try {
            GitUtils::class.java.classLoader.getResourceAsStream("git.properties").use { inputStream ->
                if (inputStream != null) {
                    info.load(inputStream)
                }
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }


    val buildVersion: String
        get() = info.getProperty("git.build.version")


    val branch: String
        get() = info.getProperty("git.branch")


    val remote: String
        get() = info.getProperty("git.remote.origin.url")


    fun getCommitId(shortID: Boolean): String {
        return if (shortID) info.getProperty("git.commit.id.abbrev") else info.getProperty("git.commit.id")
    }


    val buildUser: String
        get() = info.getProperty("git.build.user.name")


    val buildUserEmail: String
        get() = info.getProperty("git.build.user.email")


    val commitMessage: String
        get() = info.getProperty("git.commit.message.full")


    val commitTime: String
        get() = info.getProperty("git.commit.time")
}
